sdk/core/core/src/az_xml_parser.c

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <az_json_parser.h>
+
+#include <az_contract.h>
+#include <ctype.h>
+
+#include <_az_cfg.h>
+
+AZ_NODISCARD az_xml_parser az_xml_parser_create(az_span const buffer) {}
+
+AZ_NODISCARD az_result az_xml_parser_read(
+    az_xml_parser * const self,
+    az_xml_value * const out_value
+)
+{
+    AZ_CONTRACT_ARG_NOT_NULL(self);
+    AZ_CONTRACT_ARG_NOT_NULL(out_value);
+
+    return AZ_ERROR_NOT_IMPLEMENTED;
+}
+
+AZ_NODISCARD az_result az_xml_parser_read_object_member(
+    az_xml_parser * const self,
+    az_xml_member * const out_member
+)
+{
+    AZ_CONTRACT_ARG_NOT_NULL(self);
+    AZ_CONTRACT_ARG_NOT_NULL(out_member);
+
+    return AZ_ERROR_NOT_IMPLEMENTED;
+}
+
+AZ_NODISCARD az_result az_xml_parser_read_array_element(
+    az_xml_parser * const self,
+    az_xml_value * const out_value
+)
+{
+    AZ_CONTRACT_ARG_NOT_NULL(self);
+    AZ_CONTRACT_ARG_NOT_NULL(out_value);
+
+    return AZ_ERROR_NOT_IMPLEMENTED;
+}
+
+AZ_NODISCARD az_result az_xml_parser_done(
+    az_xml_parser const * const self
+)
+{
+    AZ_CONTRACT_ARG_NOT_NULL(self);
+
+    return AZ_ERROR_NOT_IMPLEMENTED;
+}
+
+AZ_NODISCARD az_result az_xml_parser_skip(
+    az_xml_parser * const self,
+    az_xml_value const value
+)
+{
+    (void)self;
```

Comment:
AZ_CONTRACT_ARG_NOT_NULL()?

Reply:
Adding all those pre-condition check in the implementation.